test/distributed/fsdp/test_wrap.py

```diff
@@ -276,6 +279,25 @@ def test_always_wrap(self):
         model = FSDP(seq, process_group=self.process_group, auto_wrap_policy=always_wrap_policy)
         TestFSDPWrap.NestedSequentialModel.verify_model_all_wrapped(self, model)
 
+    def test_transformer_auto_wrap_policy(self):
+        """
+        """
+        model = TransformerWithSharedParams(group=self.process_group)
+        my_auto_wrap_policy = functools.partial(
+            transformer_auto_wrap_policy,
+            transformer_layer_cls={TransformerEncoderLayer, TransformerDecoderLayer}
+        )
+        fsdp_model = FSDP(
+            model,
+            process_group=self.process_group,
+            auto_wrap_policy=my_auto_wrap_policy
+        )
+        self.assertTrue(isinstance(fsdp_model, FSDP))
+        for layer in fsdp_model.module.module.transformer.encoder.layers:
```

Comment:
Furthermore they should be wrapped in _individual_ FSDP units right? Can we ensure that they are in distinct FSDP units?

Reply:
yes, this for loop means they are individual FSDP units, otherwise layer will be the instance of original module name